gateway/message_handler.go

```diff
@@ -193,11 +158,92 @@ func (sc *SlackClient) handleMessageEvent(incomingChan chan<- *SlackEvent, messa
 		return
 
 	case "message_replied":
-		// As far as I can tell, this is only useful for updating the reply count of a message thread.
-		return
+		// Handle thread replies by maybe quoting the parent message if enough time has passed.
+		subMessage := messageData.SubMessage
+		if subMessage == nil {
+			log.Printf("%s message_replied with missing submessage: %+v SubMessage:%+v",
+				sc.Tag(), messageData, messageData.SubMessage)
+			return
+		}
+
+		if subMessage.Type != "message" {
+			log.Printf("%s message_replied with unexpected submessage type: %+v SubMessage:%+v",
+				sc.Tag(), messageData, messageData.SubMessage)
+			return
+		}
+
+		if !sc.shouldQuoteThreadParent(subMessage.ThreadTimestamp, messageData.Timestamp) {
+			return
+		}
+
+		sender, _, wasSenderSwapped = sc.resolveMessageEndpoints(subMessage)
+		if sender == nil || target == "" {
+			return
+		}
+
+		messageText := sc.ParseMessageText(subMessage.Text)
+		for _, attachment := range subMessage.Attachments {
+			if messageText == "" {
+				messageText = sc.ParseMessageText(attachment.Fallback)
+			} else {
+				messageText = messageText + "\n" + sc.ParseMessageText(attachment.Fallback)
+			}
+		}
+
+		// >be me
+		messageText = "><" + sender.Nick + "> " + messageText
+
+		// If we had swapped senders earlier, make sure the message reflects this swap
+		if wasSenderSwapped {
+			messageText = "[" + sc.self.Nick + "] " + messageText
+		}
+
+		// Only quote the first line of a multi-line message.
+		messageEvents := messageTextToEvents(sender, target, messageText)
+		if len(messageEvents) > 0 {
+			incomingChan <- messageTextToEvents(sender, target, messageText)[0]
+		}
 
 	default:
 		log.Printf("%s unhandled message sub-type [%v]: %+v SubMessage:%+v",
 			sc.Tag(), messageData.SubType, messageData, messageData.SubMessage)
 	}
 }
+
+func (sc *SlackClient) resolveMessageEndpoints(messageData *slack.Msg) (sender *SlackUser, target string, wasSenderSwapped bool) {
+	if messageData.User != "" {
+		var err error
+		if sender, err = sc.ResolveUser(messageData.User); err != nil {
+			log.Printf("%s could not resolve user for message [%v]: %+v", sc.Tag(), err, messageData)
+			return
+		}
+	}
+
+	if messageData.Channel != "" {
```

Comment:
Do we want to un-nest the conditional here and do what you did above?

```
if messageData.Channel == "" {
    return
}
````

and then do the rest of the logic?